### PR TITLE
[docs] Update README.md, split out HACKING.md

### PR DESCRIPTION
**Pull request by LorbusChris**
Splits out the development instructions into `docs/HACKING.md` and
updates `README.md` to align with the contents of the community
operator description on OperatorHub.

### CODE REVIEW

**Comment by LorbusChris** [comment]
cc @gmarkley-VI 
we discussed this today

File: README.md [hunk 1]
```diff
@@ -1,260 +1,64 @@
...
-You can also change the channels based on the release status of the operator.
-This command should create a new bundle image. Bundle image and operator image are two different images. 
+* OKD/OCP 4.6 cluster running on Azure or AWS, configured with hybrid OVN Kubernetes networking
 
-You should then push the newly created bundle image to the remote repository:
-```shell script
-podman push $BUNDLE_REPOSITORY:$BUNDLE_TAG
+## Usage
+Once the `openshift-windows-machine-config-operator` namespace has been created, a secret must be created containing
+the private key that will be used to access the Windows VMs. The private key should be in PEM encoded RSA format.
```
Comment (aravindhp) [inline README.md:15]: I suggest you pull in #138 here. No reason for that to be a separate PR. @PratikMahajan are you okay with that?

File: README.md [hunk 2]
```diff
@@ -1,260 +1,64 @@
...
-```
-The variables in the command should be changed to match the container image repository you wish to store the bundle in.
-You can also change the channels based on the release status of the operator.
-This command should create a new bundle image. Bundle image and operator image are two different images. 
+* OKD/OCP 4.6 cluster running on Azure or AWS, configured with hybrid OVN Kubernetes networking
 
-You should then push the newly created bundle image to the remote repository:
-```shell script
-podman push $BUNDLE_REPOSITORY:$BUNDLE_TAG
+## Usage
```
Comment (aravindhp) [inline README.md:13]: How do I run the operator?

File: README.md [hunk 3]
```diff
@@ -1,260 +1,64 @@
...
-opm index rm --from-index $INDEX_REPOSITORY:$INDEX_TAG
-```
+Below is the example of an Azure Windows MachineSet which can create Windows Machines that the WMCO can react upon.
+Please note that the windows-user-data secret will be created by the WMCO lazily when it is configuring the first
+Windows Machine. After that, the windows-user-data will be available for the subsequent MachineSets to be consumed.
+It might take around 10 minutes for the Windows VM to be configured so that it joins the cluster. Please note that
+the MachineSet should have following labels:
+* `machine.openshift.io/os-id: Windows`
+* `machine.openshift.io/cluster-api-machine-role: worker`
+* `machine.openshift.io/cluster-api-machine-type: worker`
```
Comment (aravindhp) [inline README.md:28]: I suggest using italics here. We should reserve `blah` for code and yaml blocks. There is too much `blah` in the README now so it is for aesthetic reasons :-)

**Review by aravindhp — CHANGES_REQUESTED** [review]
Thanks for doing this, @LorbusChris. I like the general direction :+1:

File: README.md (hunk 1 above)
Comment (PratikMahajan) [reply README.md:15]: Yes! I'm cool with that !

**Comment by sebsoto** [comment]
Great work, really appreciate this change, the README was a bit of a mess before.

File: README.md [hunk 4]
```diff
@@ -1,260 +1,64 @@
...
-
-You can then run the following command in the root of this git repository:
-```shell script
-operator-sdk bundle create $BUNDLE_REPOSITORY:$BUNDLE_TAG -d deploy/olm-catalog/windows-machine-config-operator/manifests \
---channels alpha --default-channel alpha --image-builder podman
-```
-The variables in the command should be changed to match the container image repository you wish to store the bundle in.
-You can also change the channels based on the release status of the operator.
-This command should create a new bundle image. Bundle image and operator image are two different images. 
+* OKD/OCP 4.6 cluster running on Azure or AWS, configured with hybrid OVN Kubernetes networking
```
Comment (sebsoto) [inline README.md:11]: Is 4.6 shipping with OVN kubernetes as default? If not we should link to instructions to create a cluster with it.

Comment (LorbusChris) [reply README.md:11]: it is not. I'll add back all the links as Aravindh suggested above

**Review by aravindhp — APPROVED** [review]
The README is so much cleaner. Thank you @LorbusChris 

/approve
